libevm/JitVM.cpp

```diff
@@ -175,9 +174,27 @@ void create(evm_result* o_result, ExtVMFace& _env, evm_message const* _msg) noex
 	}
 	else
 	{
-		o_result->code = EVM_FAILURE;
-		o_result->output_data = nullptr;
-		o_result->output_size = 0;
+		o_result->code = EVM_REVERT;
```

Comment:
This is a bit messy EVM-C concept what was around before REVERT. The idea is to pass some error messages describing what failed in the EVM. Here we try to reuse the same output buffer to pass consensus-critical REVERT output and debug messages. EVM_REVERT informs the EVM implementation that this output comes from REVERT and should not be discarded.

This long description indicates that this is probably bad idea. What do you think @axic? Maybe long list of error codes should be enough?